test/RewriteSpec.hs

```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module RewriteSpec where
+
+import Text.RawString.QQ (r)
+import qualified Data.Text as T
+import qualified Data.Text.IO as T
+import OurPrelude
+import qualified Rewrite
+import Test.Hspec
+import qualified Utils
+import qualified Polysemy.Output as Output
+import qualified Polysemy.Error as Error
+import qualified Process
+import qualified File
+
+main :: IO ()
+main = hspec spec
+
+spec :: Spec
+spec = do
+  describe "Rewrite.quotedUrls" do
+    it "quotes an unquoted meta.homepage URL" do
+      let options = Utils.Options False "" False False
+      let updateEnv = Utils.UpdateEnv "inadyn" "2.5" "2.6" Nothing options
+      let rwArgs = Rewrite.Args updateEnv "inadyn" undefined noQuotedHompageNix
+      (logs, (newContents, result)) <-
+        (runFinal
+          . embedToFinal @IO
+          . Output.runOutputList
+          . File.runPure [noQuotedHompageNix]
+          . Process.runPure ["\"http://troglobit.com/project/inadyn/\""]
+          . Error.errorToIOFinal
+          $ Rewrite.quotedUrls rwArgs) :: IO ([Text], ([Text], Either Text (Maybe Text)))
+      T.putStrLn $ T.unlines logs
+      result `shouldBe` Right (Just "Quoted meta.homepage for [RFC 45](https://github.com/NixOS/rfcs/pull/45)")
+      head newContents `shouldSatisfy` ("\"http://troglobit.com/project/inadyn/\"" `T.isInfixOf`)
+
+
+noQuotedHompageNix :: Text
```

Comment:
We might want to do this by convention, where we have `quoted_homepage_bad.nix` (the original) and `quoted_homepage_good.nix` (the expected result), and then just assert that 

`rewrite quoted_homepage_bad.nix == quoted_homepage_good.nix`.  

This would prevent our test cases from getting enormous as we add more of them, and also lets you run `diff` on the files at the cmdline to see what you expect the rewrite rule to actually do.

Inspiration: https://github.com/nix-community/nixpkgs-fmt/tree/master/test_data

Reply:
Did this!